internal/constraints/constraints.go

```diff
@@ -82,8 +82,7 @@ type projectable interface {
 	Path() string
 	Dir() string
 	URL() string
-	LegacyCommitID() string // for commitmediator.Get
-	StripLegacyCommitID() error
+	LegacyCommitID() string
```

Comment:
Nit: I'd change the comment to be `// for localcommit.Get` since it's not clear why this interface needs this method.

Reply:
I dropped it because it's not necessary. All we need to know is that it's legacy. What code it's used by is what tooling is for. Case in point; all comments like this were essentially invalidated by this PR, and tooling doesn't update comments..